packages/components/header/template.njk

```diff
@@ -81,6 +81,10 @@
           <a class="nhsuk-header__account-link" href="{{ item.href }}">
             {{ _accountItem(item) }}
           </a>
+        {% elif item.action %}
+          <button class="nhsuk-header__account-button" formaction="{{ item.action }}" formmethod="{{ item.method or "post" }}">
```

Comment:
Must admit I’ve never come across the `formaction` or `formmethod` attributes for buttons, but they’re in the spec and look like they’re well supported?

Does still need a separate `form` element though, as you say.

I did wonder if it’d be simpler for the nunjucks to generate a separate form tag for each of the buttons (in most cases it’d be be log out?) - but then I guess in production you might also have extra hidden inputs for cross-site request forgery (CSRF) tokens or whatever?

Reply:
Precisely. This is the corresponding (Rails) implementation we currently have on Mavis:

```html
<span class="app-header__account-item">
  <form class="button_to" method="post" action="/logout">
    <input type="hidden" name="_method" value="delete" autocomplete="off">
    <button class="app-header__account-button" type="submit">Log out</button>
    <input type="hidden" name="authenticity_token" value="XXX" autocomplete="off">
  </form>
</span>
```

Unsure if adding a `<form>` in the Nunjucks macro if any button includes an action is helpful or not. As Nunjucks is mostly used by prototypes, maybe it is?